projects/media-viewer/src/lib/icp/icp.directive.ts

```diff
@@ -0,0 +1,97 @@
+import { Directive, Input, OnDestroy, OnInit } from '@angular/core';
+import { ToolbarEventService } from '../toolbar/toolbar-event.service';
+import { select, Store } from '@ngrx/store';
+import * as fromIcpSelectors from '../store/selectors/icp.selectors';
+import * as fromIcpActions from '../store/actions/icp.action';
+import { Subscription } from 'rxjs';
+import { IcpUpdateService } from './icp-update.service';
+import { ViewerEventService } from '../viewers/viewer-event.service';
+import { IcpPresenterService } from './icp-presenter.service';
+import { IcpFollowerService } from './icp-follower.service';
+import { filter, take } from 'rxjs/operators';
+import { IcpParticipant, IcpState } from './icp.interfaces';
+
+
+@Directive({
+  selector: '[inCourtPresentation]'
+})
+export class IcpDirective implements OnInit, OnDestroy {
+
+  @Input() caseId: string;
+
+  client: IcpParticipant;
+  presenter: IcpParticipant;
+  isPresenter: boolean;
+
+  subscription: Subscription;
+  sessionSubscription: Subscription;
+
+  constructor(private readonly toolbarEvents: ToolbarEventService,
```

Comment:
https://angular.io/guide/attribute-directives
this approach is going against the angular guidelines. 
directives are meant to use only for manipulating elements - but i like your thinking

Reply:
we can argue that we're manipulating the component, which is the pdf-viewer